test/reporters/base.spec.js

```diff
@@ -160,6 +168,46 @@ describe('Base reporter', function () {
     });
   });
 
+  describe('unified diff reporter', function () {
+    it('should separate diff hunks by two dashes', function () {
+      var err = new Error('test');
+      var errOut;
+
+      err.actual = createElements({ from: 2, to: 11 });
+      err.expected = createElements({ from: 1, to: 10 });
+      err.showDiff = true;
+      var test = makeTest(err);
+
+      Base.inlineDiffs = false;
+      Base.list([test]);
+
+      errOut = stdout.join('\n');
+
+      var regexesToMatch = [
+        /\[/,
+        /\+ {2}"element 1"/,
+        /"element 2"/,
+        /"element 3"/,
+        /"element 4"/,
+        /"element 5"/,
+        /--/,
+        /"element 7"/,
+        /"element 8"/,
+        /"element 9"/,
+        /"element 10"/,
+        /- {2}"element 11"/,
+        /]/,
+        /test/,
+        /expected/,
+        /actual/
+      ];
+
+      regexesToMatch.forEach(function (aRegex) {
+        errOut.should.match(aRegex);
+      });
```

Comment:
When I PR - I try to avoid optimizations like these and just stick with the patterns in the existing codebase.  I too thought this pattern was weird, but it's [what's in the code](https://github.com/mochajs/mocha/blob/master/test/reporters/base.spec.js#L149-L159).

If you really want me to make it different for this particular test I will, but I suggest refactoring them all in a separate PR for more accurate test coverage.